pkg/node/address.go

```diff
@@ -216,30 +220,55 @@ func GetIPv6AllocRange() *cidr.CIDR {
 	return ipv6AllocRange
 }
 
-// SetExternalIPv4 sets the external IPv4 node address. It must be reachable on the network.
-func SetExternalIPv4(ip net.IP) {
-	ipv4ExternalAddress = ip
+// SetIPv4 sets the IPv4 node address. It must be reachable on the network.
+// It is set based on the following priority:
+// - NodeInternalIP
+// - NodeExternalIP
+// - other IP address type
+func SetIPv4(ip net.IP) {
+	ipv4Address = ip
 }
 
-// GetExternalIPv4 returns the external IPv4 node address
-func GetExternalIPv4() net.IP {
-	return ipv4ExternalAddress
+// GetIPv4 returns one of the IPv4 node address available with the following
+// priority:
+// - NodeInternalIP
+// - NodeExternalIP
+// - other IP address type.
+// It must be reachable on the network.
+func GetIPv4() net.IP {
+	return ipv4Address
+}
+
+// SetInternalIPv4Router sets the cilium internal IPv4 node address, it is allocated from the node prefix.
+// This must not be conflated with k8s internal IP as this IP address is only relevant within the node.
+// Cilium doesn't respond to ARP for this IP on the native devices, but only on the `cilium_host` device.
```

Comment:
While I believe that this statement is correct today, I don't think it's a good idea to document it here since that behaviour would be subject to entirely different code so this comment could easily be wrong or get out of date as changes are made to the codebase.

```suggestion
```

Reply:
Fixed,